src/style.rs

```diff
@@ -176,3 +235,28 @@ impl FromStr for Margin {
         }
     }
 }
+
+impl FromStr for Radius {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let values = s
+            .split(' ')
+            .map(|s| s.trim())
+            .filter(|s| !s.is_empty())
+            .map(|s| s.parse::<f32>().map_err(|_| "invalid float value"))
```

Comment:
It seems it is also possible to parse `f32::{INF,NEG_INF,NAN}` which barely make sense there. So does negative values

Reply:
Good catch. I have now added a check prohibiting parsing non-finite float values for `Radius`, `Margin` and `Padding` as well as one allowing only positive values for `Radius`.